clang/include/clang/ExtractAPI/Serialization/SerializerBase.h

```diff
@@ -28,11 +28,78 @@ struct APISerializerOption {
 };
 
 /// The base interface of serializers for API information.
-class APISerializer {
+template <typename Derived> class APISerializer {
 public:
   /// Serialize the API information to \p os.
   virtual void serialize(raw_ostream &os) = 0;
 
+  virtual void traverseGlobalFunctionRecords() {
+    for (const auto &globalFunction : API.getGlobalFunctions())
+      static_cast<Derived *>(this)->visitGlobalFunctionRecord(
+          std::move(*globalFunction.second));
```

Comment:
This doesn't have the semantics you might think it does, you are using `std::move` on a const reference to an `APIRecord`, but the reference is const so what winds up happening here is a copy and then a move of the copied object.. Since `visitGlobalFunction` doesn't need it's own copy of the object just pass it a const reference with `*globalFunction.second` without the move.